### PR TITLE
fix: Error handling for the state log download failure

### PR DESCRIPTION
**Pull request by NiranjanaBinoy**
## **Description**
Display error message if the `exportAsFile()` for state log download in advanced tab in the settings fails.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26999?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2875

## **Manual testing steps**

1. Throw an error artificially in the `exportAsFile()` state logs download function
2. `Failed to download state log.` message is diaplayed

### **Before**
Screenshot 2024-11-01 at 10 21 23 AM

### **After**

Screenshot 2024-11-01 at 10 34 37 AM

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/settings/advanced-tab/advanced-tab.component.test.js [hunk 1]
```diff
@@ -21,6 +25,43 @@ jest.mock('../../../store/actions.ts', () => {
...
+// jest.mock('../../../helpers/utils/export-utils', () => ({
+//   exportAsFile: () => mockExportAsFile
+// }));
+
+// jest.mock('../../../helpers/utils/export-utils', () => {
+//   return {
+//     exportAsFile: () => mockExportAsFile,
+//   };
+// });
+// jest.mock('../../../helpers/utils/export-utils');
```
Comment (legobeat) [inline ui/pages/settings/advanced-tab/advanced-tab.component.test.js:56]: Should these commented mocks stay or go?

File: ui/pages/settings/advanced-tab/advanced-tab.component.test.js [hunk 2]
```diff
@@ -105,4 +146,24 @@ describe('AdvancedTab Component', () => {
...
+    it('should call exportAsFile when the toggle button is clicked', () => {
+      // const exportSpy = jest.spyOn(exportUtil, 'exportAsFile')
+      // const exportAsFileMocked = jest.mocked(exportAsFile);
+      const { queryByTestId } = renderWithProvider(<AdvancedTab />, mockStore);
+      const stateLogButton = queryByTestId(
+        'advanced-setting-state-logs-button',
+      );
+      fireEvent.click(stateLogButton);
+      expect(exportUtil.exportAsFile).toHaveBeenCalled();
+      // await expect(mockExportAsFile).toHaveBeenCalled();
```
Comment (legobeat) [inline ui/pages/settings/advanced-tab/advanced-tab.component.test.js:166]: Commented code

File: ui/pages/settings/advanced-tab/advanced-tab.component.test.js [hunk 3]
```diff
@@ -105,4 +146,24 @@ describe('AdvancedTab Component', () => {
...
+
+    it('should call exportAsFile when the toggle button is clicked', () => {
+      // const exportSpy = jest.spyOn(exportUtil, 'exportAsFile')
+      // const exportAsFileMocked = jest.mocked(exportAsFile);
+      const { queryByTestId } = renderWithProvider(<AdvancedTab />, mockStore);
+      const stateLogButton = queryByTestId(
+        'advanced-setting-state-logs-button',
+      );
+      fireEvent.click(stateLogButton);
+      expect(exportUtil.exportAsFile).toHaveBeenCalled();
```
Comment (legobeat) [inline ui/pages/settings/advanced-tab/advanced-tab.component.test.js:165]: This test is failing? 
https://github.com/MetaMask/metamask-extension/actions/runs/11040557627/job/30668816032#step:4:7979

**Review by legobeat — CHANGES_REQUESTED** [review]

File: ui/pages/settings/advanced-tab/advanced-tab.component.test.js (hunk 1 above)
Comment (NiranjanaBinoy) [reply ui/pages/settings/advanced-tab/advanced-tab.component.test.js]: File is updated now

File: ui/pages/settings/advanced-tab/advanced-tab.component.test.js (hunk 2 above)
Comment (NiranjanaBinoy) [reply ui/pages/settings/advanced-tab/advanced-tab.component.test.js]: File is updated now

**Review by DDDDDanica — DISMISSED** [review]

**Comment by DDDDDanica** [comment]
LGTM ! Let me know when u resolve the conflicts !

**Review by DDDDDanica — APPROVED** [review]

**Review by hjetpoluru — APPROVED** [review]
LGTM

**Pull request merged**